pkg/reconciler/identity/reconciler.go

```diff
@@ -102,7 +103,7 @@ func (i *Identity) ReconcileWorkloadIdentity(ctx context.Context, projectID stri
 	}
 
 	// Add iam policy binding to GCP ServiceAccount.
-	if err := i.addIamPolicyBinding(ctx, projectID, googleServiceAccount, kServiceAccount); err != nil {
+	if err := i.addIamPolicyBinding(ctx, projectID, identityNames.GoogleServiceAccountName, kServiceAccount); err != nil {
```

Comment:
(Feel free to pushback) If reasonable, pass in `identityNames`, rather than the two strings.